code/game/objects/items/stacks/stack.dm

```diff
@@ -10,156 +10,171 @@
-			t1+="<br>"
+		if(i > 1 && !isnull(recipe_list[i - 1]))
+			t1 += "<br>"
 
 		if(istype(E, /datum/stack_recipe_list))
 			var/datum/stack_recipe_list/srl = E
-			if(src.amount >= srl.req_amount)
-				t1 += "<a href='?src=\ref[src];sublist=[i]'>[srl.title] ([srl.req_amount] [src.singular_name]\s)</a>"
+			if(amount >= srl.req_amount)
+				t1 += "<a href='?src=\ref[src];sublist=[i]'>[srl.title] ([srl.req_amount] [singular_name]\s)</a>"
 			else
-				t1 += "[srl.title] ([srl.req_amount] [src.singular_name]\s)<br>"
+				t1 += "[srl.title] ([srl.req_amount] [singular_name]\s)<br>"
 
 		if(istype(E, /datum/stack_recipe))
 			var/datum/stack_recipe/R = E
 			var/max_multiplier = round(src.amount / R.req_amount)
-			var/title as text
+			var/title
 			var/can_build = 1
-			can_build = can_build && (max_multiplier>0)
+			can_build = can_build && (max_multiplier > 0)
 			/*
 			if(R.one_per_turf)
 				can_build = can_build && !(locate(R.result_type) in usr.loc)
 			if(R.on_floor)
 				can_build = can_build && istype(usr.loc, /turf/simulated/floor)
 			*/
-			if(R.res_amount>1)
-				title+= "[R.res_amount]x [R.title]\s"
+			if(R.res_amount > 1)
+				title += "[R.res_amount]x [R.title]\s"
 			else
-				title+= "[R.title]"
-			title+= " ([R.req_amount] [src.singular_name]\s)"
+				title += "[R.title]"
+			title += " ([R.req_amount] [src.singular_name]\s)"
 			if(can_build)
-				t1 += text("<A href='?src=\ref[src];sublist=[recipes_sublist];make=[i]'>[title]</A>  ")
+				t1 += "<A href='?src=\ref[src];sublist=[recipes_sublist];make=[i]'>[title]</A>  "
 			else
-				t1 += text("[]", title)
+				t1 += "[title]"
 				continue
 			if(R.max_res_amount>1 && max_multiplier>1)
-				max_multiplier = min(max_multiplier, round(R.max_res_amount/R.res_amount))
+				max_multiplier = min(max_multiplier, round(R.max_res_amount / R.res_amount))
 				t1 += " |"
 				var/list/multipliers = list(5,10,25)
 				for(var/n in multipliers)
 					if(max_multiplier>=n)
-						t1 += " <A href='?src=\ref[src];make=[i];multiplier=[n]'>[n*R.res_amount]x</A>"
+						t1 += " <A href='?src=\ref[src];make=[i];multiplier=[n]'>[n * R.res_amount]x</A>"
 				if(!(max_multiplier in multipliers))
-					t1 += " <A href='?src=\ref[src];make=[i];multiplier=[max_multiplier]'>[max_multiplier*R.res_amount]x</A>"
+					t1 += " <A href='?src=\ref[src];make=[i];multiplier=[max_multiplier]'>[max_multiplier * R.res_amount]x</A>"
 
 	var/datum/browser/popup = new(user, "stack", name, 400, 400)
 	popup.set_content(t1)
 	popup.open(0)
 	onclose(user, "stack")
-	return
 
 /obj/item/stack/Topic(href, href_list)
 	..()
-	if((usr.restrained() || usr.stat || usr.get_active_hand() != src))
-		return
+	if(usr.incapacitated() || usr.get_active_hand() != src)
+		return 0
 
 	if(href_list["sublist"] && !href_list["make"])
 		list_recipes(usr, text2num(href_list["sublist"]))
 
 	if(href_list["make"])
-		if(src.amount < 1) del(src) //Never should happen
+		if(amount < 1)
+			del(src) //Never should happen
```

Comment:
It's beyond the scope of the PR to literally change a single `del` to `qdel(src)` and have. 

```
/obj/item/stack/Destroy()
	..()
	return QDEL_HINT_HARDDEL_NOW
```